src/olympia/addons/admin.py

```diff
@@ -83,8 +85,13 @@ def clean(self):
         path = None
         try:
             path = self.data.get('path')
-            path = ('/' if not path.startswith('/') else '') + path
-            resolve(path)
+            site = settings.SITE_URL
+            if (models.ReplacementAddon.is_external_path(path) and
+                    not path.startswith(site)):
```

Comment:
It looks like a site link causes the resolve below to blow up? It might be nicer to do a specific validation error with a  message so there's some feedback to the user to explain why it's gone wrong.

Reply:
It wouldn't blow up, but I didn't want to allow it because a full link to an AMO addon page wouldn't get the GUID lookup for the API that TAAR uses.  I can return a specific error though.